src/VecSim/algorithms/brute_force/brute_force.h

```diff
@@ -201,10 +178,7 @@ void BruteForceIndex<DataType, DistType>::removeVector(idType id_to_delete) {
     labelType last_idx_label = getVectorLabel(last_idx);
 
     // Get last vector data.
-    DataBlock &last_vector_block = vectorBlocks.back();
-    assert(&last_vector_block == &getVectorVectorBlock(last_idx));
-
-    void *last_vector_data = last_vector_block.removeAndFetchLastElement();
+    const char *last_vector_data = vectors->getElement(last_idx);
```

Comment:
this can be inside the if, no?
or replaced with pop + fetch + remove `vectors->removeElement(last_idx);`